solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClientBase.java

```diff
@@ -340,6 +340,20 @@ protected void setParser(ResponseParser parser) {
 
   protected abstract void updateDefaultMimeTypeForParser();
 
+  /**
+   * Executes a SolrRequest using the provided URL to temporarily override any "base URL" currently
+   * used by this client
+   *
+   * @param baseUrl a URL to a root Solr path (i.e. "/solr") that should be used for this request
+   * @param solrRequest the SolrRequest to send
+   * @param collection an optional collection or core name used to override the client's "default
+   *     collection". May be 'null' for any requests that don't require a collection or wish to rely
+   *     on the client's default
+   */
+  public abstract NamedList<Object> requestWithBaseUrl(
```

Comment:
from the other PR but it's very nice here.  I'd like to see that one merged ASAP.